ldk/javascript/src/whisper/form.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { Components, Form, Whisper, WhisperComponentType } from '.';
+
+export class LdkForm {
+    children: Array<Components>;
+
+    componentState: Map<string, any>;
+
+    constructor(children: Array<Components>) {
+        this.componentState = new Map<string, any>();
+        this.children = children;
+
+        this.setInitialComponentState(children);
+        this.registerListeners(children);
+    }
+
+    getComponentState(): Map<string, any> {
+        return this.componentState;
+    }
+
+    private isOnChangeInput(child: any): boolean {
+        return (child.onChange !== undefined && child.name !== undefined);
+    }
+
+    private isOnSelectInput(child: any): boolean {
+        return (child.onSelect !== undefined && child.name !== undefined);
+    }
+
+    private setInitialComponentState(children: Array<any>) {
+        children.forEach((child: any) => {
+            if (this.isOnChangeInput(child)) {
+                this.componentState = this.componentState.set(child.name, child.value);
+            } else if (this.isOnSelectInput(child)) {
+                this.componentState = this.componentState.set(child.name, child.selected);
+            }
+        });
+    }
+
+    private registerListeners(children: Array<any>) {
```

Comment:
Should it be Array<Components> instead of any?

Reply:
I am not sure what you mean.